### ISSUE TITLE
CloudFoundryOperations.applications().list() fails in case of User Provided Service

### ISSUE THREAD

**Issue by making** [cloudfoundry/cf-java-client#354]
`cloudFoundryOperations.applications().list()` seems to fail when User Provided Service exists.

The following is error log.

``` console
2016-01-15 23:57:29.524  INFO 11181 --- [           main] com.example.DemoApplication              : Started DemoApplication in 11.14 seconds (JVM running for 12.132)
2016-01-15 23:57:29.537 DEBUG 11181 --- [client-spring-1] o.s.s.oauth2.client.OAuth2RestTemplate   : Setting request Accept header to [application/json, application/*+json]
2016-01-15 23:57:29.985 DEBUG 11181 --- [client-spring-1] o.s.s.oauth2.client.OAuth2RestTemplate   : GET request for "https://api.run.pivotal.io/v2/spaces/****/summary" resulted in 200 (OK)
2016-01-15 23:57:29.986 DEBUG 11181 --- [client-spring-1] o.s.s.oauth2.client.OAuth2RestTemplate   : Reading [class org.cloudfoundry.client.v2.spaces.GetSpaceSummaryResponse] as "application/json;charset=utf-8" using [org.springframework.http.converter.json.MappingJackson2HttpMessageConverter@3288ac44]
2016-01-15 23:57:30.187 ERROR 11181 --- [client-spring-1] reactor.core.processor.ProcessorGroup    : Unrouted exception

org.springframework.http.converter.HttpMessageNotReadableException: Could not read document: Can not deserialize instance of java.lang.String out of START_OBJECT token
 at [Source: java.io.PushbackInputStream@79c8eb38; line: 676, column: 27] (through reference chain: org.cloudfoundry.client.v2.spaces.GetSpaceSummaryResponse["services"]->java.util.ArrayList[1]->org.cloudfoundry.client.v2.serviceinstances.ServiceInstance["last_operation"]); nested exception is com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.lang.String out of START_OBJECT token
 at [Source: java.io.PushbackInputStream@79c8eb38; line: 676, column: 27] (through reference chain: org.cloudfoundry.client.v2.spaces.GetSpaceSummaryResponse["services"]->java.util.ArrayList[1]->org.cloudfoundry.client.v2.serviceinstances.ServiceInstance["last_operation"])
at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.readJavaType(AbstractJackson2HttpMessageConverter.java:224) ~[spring-web-4.2.4.RELEASE.jar:4.2.4.RELEASE]
at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.read(AbstractJackson2HttpMessageConverter.java:208) ~[spring-web-4.2.4.RELEASE.jar:4.2.4.RELEASE]
at org.springframework.web.client.HttpMessageConverterExtractor.extractData(HttpMessageConverterExtractor.java:95) ~[spring-web-4.2.4.RELEASE.jar:4.2.4.RELEASE]
at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:599) ~[spring-web-4.2.4.RELEASE.jar:4.2.4.RELEASE]
at org.springframework.security.oauth2.client.OAuth2RestTemplate.doExecute(OAuth2RestTemplate.java:128) ~[spring-security-oauth2-2.0.8.RELEASE.jar:na]
at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:572) ~[spring-web-4.2.4.RELEASE.jar:4.2.4.RELEASE]
at org.springframework.web.client.RestTemplate.getForObject(RestTemplate.java:280) ~[spring-web-4.2.4.RELEASE.jar:4.2.4.RELEASE]
at org.cloudfoundry.client.spring.util.AbstractSpringOperations$3.apply(AbstractSpringOperations.java:110) ~[cloudfoundry-client-spring-2.0.0.BUILD-20160115.114537-295.jar:2.0.0.BUILD-SNAPSHOT]
at org.cloudfoundry.client.spring.util.AbstractSpringOperations$3.apply(AbstractSpringOperations.java:101) ~[cloudfoundry-client-spring-2.0.0.BUILD-20160115.114537-295.jar:2.0.0.BUILD-SNAPSHOT]
at org.cloudfoundry.client.spring.util.AbstractSpringOperations$2$1.accept(AbstractSpringOperations.java:86) ~[cloudfoundry-client-spring-2.0.0.BUILD-20160115.114537-295.jar:2.0.0.BUILD-SNAPSHOT]
at org.cloudfoundry.client.spring.util.AbstractSpringOperations$2$1.accept(AbstractSpringOperations.java:80) ~[cloudfoundry-client-spring-2.0.0.BUILD-20160115.114537-295.jar:2.0.0.BUILD-SNAPSHOT]
at reactor.core.publisher.FluxSession$YieldingReactiveSession.request(FluxSession.java:77) ~[reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.processor.ProcessorGroup$ProcessorBarrier.doRequest(ProcessorGroup.java:773) ~[reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.processor.ProcessorGroup$WorkProcessorBarrier.run(ProcessorGroup.java:1033) ~[reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.processor.ProcessorGroup$WorkProcessorBarrier.request(ProcessorGroup.java:1058) ~[reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.publisher.MonoNext$MonoNextSubscriber.request(MonoNext.java:107) ~[reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.publisher.FluxFlatMap$BufferSubscriber.onSubscribe(FluxFlatMap.java:673) ~[reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.publisher.MonoNext$MonoNextSubscriber.onSubscribe(MonoNext.java:69) ~[reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.processor.ProcessorGroup$WorkProcessorBarrier$1.run(ProcessorGroup.java:1005) ~[reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.processor.ProcessorGroup$TaskSubscriber.onNext(ProcessorGroup.java:1167) [reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.processor.ProcessorGroup$TaskSubscriber.onNext(ProcessorGroup.java:1142) [reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at reactor.core.processor.RingBufferWorkProcessor$QueueSubscriber.run(RingBufferWorkProcessor.java:828) [reactor-core-2.5.0.BUILD-20160114.182314-64.jar:na]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_60]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_60]
at java.lang.Thread.run(Thread.java:745) [na:1.8.0_60]
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.lang.String out of START_OBJECT token
 at [Source: java.io.PushbackInputStream@79c8eb38; line: 676, column: 27] (through reference chain: org.cloudfoundry.client.v2.spaces.GetSpaceSummaryResponse["services"]->java.util.ArrayList[1]->org.cloudfoundry.client.v2.serviceinstances.ServiceInstance["last_operation"])
at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:148) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.DeserializationContext.mappingException(DeserializationContext.java:857) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:62) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:11) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:463) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:378) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1099) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:296) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:133) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:217) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:25) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:463) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:378) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1099) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:296) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:133) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3736) ~[jackson-databind-2.6.4.jar:2.6.4]
at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2810) ~[jackson-databind-2.6.4.jar:2.6.4]
at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.readJavaType(AbstractJackson2HttpMessageConverter.java:221) ~[spring-web-4.2.4.RELEASE.jar:4.2.4.RELEASE]
... 24 common frames omitted
```

Registered services are as follows. `aaa-redis` is a user provided service, which doesn't have some attributes like `last_operation`, `service_plan`.

``` json
      {
         "guid": "****",
         "name": "aaa-redis",
         "bound_app_count": 0
      },
      {
         "guid": "****",
         "name": "bbb-redis",
         "bound_app_count": 1,
         "last_operation": {
            "type": "create",
            "state": "succeeded",
            "description": "",
            "updated_at": null,
            "created_at": "2015-12-27T18:09:16Z"
         },
         "dashboard_url": "https://cloudfoundry.appdirect.com/api/custom/cloudfoundry/v2/sso/start?serviceUuid=****",
         "service_plan": {
            "guid": "1c5a2ed5-35ea-4786-9ca7-8bc13593fae3",
            "name": "30mb",
            "service": {
               "guid": "99b6e48d-f319-4a72-b854-a6e43eea9c3c",
               "label": "rediscloud",
               "provider": null,
               "version": null
            }
         }
      }
```

**Comment by nebhale** [comment]
@making We independently figured this out earlier today.  The problem is that [our documentation](http://apidocs.cloudfoundry.org/227/spaces/get_space_summary.html) gives us no idea what the type of `last_operation` is.  Now that we know, we'll get on fixing it.

**Comment by nebhale** [comment]
Resolved by 9581a201865b381f434f735cadf558d04063a3bc
